llvm/lib/Target/RISCV/RISCVISelLowering.cpp

```diff
@@ -12764,6 +12788,15 @@ struct NodeExtensionHelper {
   /// Check if \p Root supports any extension folding combines.
   static bool isSupportedRoot(const SDNode *Root) {
     switch (Root->getOpcode()) {
+    case ISD::ADD:
+    case ISD::SUB:
+    case ISD::MUL: {
+      EVT VT0 = Root->getOperand(0).getValueType();
```

Comment:
It should be sufficient to check `return N->getValueType().isScalableVector()` The operands of ISD::ADD/SUB/MUL must all have the same as the result.

Reply:
Do we need o check that the scalable type is legal?

Reply:
You are right, It will be safer to do this, I will fix it soon.

Reply:
I have done it.